### PR TITLE
Surveys: HTML content for answers and description

### PR DESCRIPTION
**Pull request by digorath**
- added new directory survey in components
  - moved SurveyAnswer there
  - created new component SurveyText
    - that component renders an HTMLView if the text contains html
    - otherwise it renders a RegularText
- added new width prop to htmlView to manipulate max width

---

SVA-264

---

## How to test:
* [x] Android portrait mode
* [x] iOS portrait mode
* [x] Android landscape mode
* [x] iOS landscape mode

## Screenshots:

Simulator Screen Shot - iPhone 12 Pro - 2021-11-23 at 11 29 40
Simulator Screen Shot - iPhone 12 Pro - 2021-11-23 at 11 29 43
Simulator Screen Shot - iPhone 12 Pro - 2021-11-23 at 11 45 25
Simulator Screen Shot - iPhone 12 Pro - 2021-11-23 at 11 29 36

### CODE REVIEW

File: src/components/HtmlView.js [hunk 1]
```diff
@@ -61,15 +61,15 @@ const htmlConfig = {
...
 
-  if (width > consts.DIMENSIONS.FULL_SCREEN_MAX_WIDTH) {
+  if (calculatedWidth > consts.DIMENSIONS.FULL_SCREEN_MAX_WIDTH) {
     // image width should be only 70% on wider screens, as there are 15% padding on each side
-    width = width * 0.7;
+    calculatedWidth = calculatedWidth * 0.7;
   }
 
-  const maxWidth = width - 2 * normalize(14); // width of an image minus paddings
+  const maxWidth = calculatedWidth - 2 * normalize(14); // width of an image minus paddings
```
Comment (donni106) [inline src/components/HtmlView.js:72]: |before|after|
|---|---|
|IMG_0010|IMG_0011|

File: src/components/survey/SurveyText.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+};
+
+export const SurveyText = ({ content, italic, width }: Props) => {
+  if (!content?.length) {
+    return null;
+  }
+
+  return containsHtml(content) ? (
+    // @ts-expect-error HTMLView uses memo, which is a generic type (used in js), which causes issues with type inference
+    <HtmlView html={content} tagsStyles={{ img: { alignSelf: 'flex-start' } }} width={width} />
```
Comment (donni106) [inline src/components/survey/SurveyText.tsx:20]: |before|after|
|---|---|
|IMG_0012|IMG_0013|

**Review by donni106 — APPROVED** [review]

File: src/components/HtmlView.js (hunk 1 above)
Comment (digorath) [reply src/components/HtmlView.js:72]: :+1:

File: src/components/survey/SurveyText.tsx (hunk 2 above)
Comment (digorath) [reply src/components/survey/SurveyText.tsx:20]: :+1:

**Pull request merged**